### PR TITLE
feat: add scroll interaction

### PR DESCRIPTION
**Pull request by marcossevilla**
## Description

Small POC for a new scroll interaction using `tester.fling`.

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: lib/src/interactions.dart [hunk 1]
```diff
@@ -26,3 +26,18 @@ Interaction longPress(Finder finder) => (WidgetTester tester) async {
...
+Interaction scroll(
+  Finder finder, {
+  required Offset offset,
+  double speed = kMinFlingVelocity,
+}) =>
+    (WidgetTester tester) async {
+      final elements = finder.evaluate();
+      return () async {
+        for (final element in elements) {
+          await tester.fling(find.byWidget(element.widget), offset, speed);
```
[the remark this answers is not in the document]
Comment (marcossevilla) [reply lib/src/interactions.dart:40]: yeah totally, I just took some inspiration from `tester.pressAll` but we can refactor it

File: lib/src/interactions.dart [hunk 2]
```diff
@@ -6,9 +6,9 @@ import 'package:flutter_test/flutter_test.dart';
 /// An interaction to perform while rendering a golden test. Returns
 /// an asynchronous callback that should be called to cleanup when
 /// the golden test completes.
-typedef Interaction = Future<AsyncCallback?> Function(WidgetTester);
+typedef Interaction = Future<AsyncCallback> Function(WidgetTester);
```
Comment (marcossevilla) [inline lib/src/interactions.dart:9]: not sure if there was any case in which we wanted a nullable callback, so I made it non-nullable as the comment says that it **should** be called when the test completes

File: lib/src/interactions.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (definitelyokay) [reply lib/src/interactions.dart]: ooh, maybe pressAll can be optimized too

File: lib/src/interactions.dart (hunk 2 above)
Comment (definitelyokay) [reply lib/src/interactions.dart:9]: doing this would make this a breaking api change technically

File: lib/src/interactions.dart [hunk 3]
```diff
@@ -26,3 +26,23 @@ Interaction longPress(Finder finder) => (WidgetTester tester) async {
...
+        await Future.wait([
+          for (final element in elements)
+            tester.fling(
+              find.byWidget(element.widget),
+              offset,
+              speed,
+            ),
+        ]);
+      };
+    };
```
Comment (Kirpal) [inline lib/src/interactions.dart:48]:
```suggestion
    (WidgetTester tester) async {
      final elements = finder.evaluate();
      await Future.wait([
        for (final element in elements)
          tester.fling(
            find.byWidget(element.widget),
            offset,
            speed,
          ),
      ]);
      return null;
    };
```

**Review by Kirpal — CHANGES_REQUESTED** [review]
Looks cool, just left a fix for how `cleanup` is used.

File: test/src/interactions_test.dart [hunk 4]
```diff
@@ -72,4 +73,35 @@ void main() {
...
+      find.byType(Scrollable),
+      offset: const Offset(0, -dragOffset),
+    )(tester);
+
+    await cleanup?.call();
+
+    final viewportFinder = find.byType(Viewport);
+    final viewport = tester.renderObject(viewportFinder) as RenderViewport;
+
+    expect(viewport.offset.pixels, dragOffset);
```
Comment (definitelyokay) [inline test/src/interactions_test.dart:105]: beautiful 👏

**Review by definitelyokay — APPROVED** [review]

**Review by Kirpal — APPROVED** [review]

**Pull request merged**